junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ExtensionContext.java

```diff
@@ -259,6 +264,19 @@ default void publishReportEntry(String key, String value) {
 	 */
 	interface Store {
 
+		/**
+		 * Classes implementing this interface indicate that they want to close
+		 * some underlying resources when the store is closed.
+		 */
+		interface CloseableResource {
```

Comment:
Why not use `java.lang.AutoCloseable` instead?

Reply:
It might break existing code -- as you may already store instances implementing `java.lang.AutoCloseable` which would be closed after an upgrade to 5.1.

Reply:
Well, yes, but then again -- if those are implemented to be idempotent as the Javadoc clearly suggests/recommends, then repeated calls to `.close()` should not be a problem...

Reply:
...the first call now happening is the breaking change. JUnit 5 does not close any stored object at the moment. It is up to the "storer" to close it -- which could be more than just calling `close()`.